dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/RenormalizationProcessor.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.core.models.IVariant;
+import uk.ac.ebi.eva.commons.core.models.IVariantSourceEntry;
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
+import uk.ac.ebi.eva.dbsnpimporter.sequence.FastaSequenceReader;
+
+import java.util.stream.Collectors;
+
+public class RenormalizationProcessor implements ItemProcessor<Variant, Variant> {
+
+    private static final Logger logger = LoggerFactory.getLogger(RenormalizationProcessor.class);
+
+    private FastaSequenceReader fastaSequenceReader;
+
+    public RenormalizationProcessor(FastaSequenceReader fastaSequenceReader) {
+        this.fastaSequenceReader = fastaSequenceReader;
+    }
+
+    @Override
+    public Variant process(Variant variant) throws Exception {
+        if (isAmbiguous(variant)) {
+            return renormalize(variant);
+        }
+        return variant;
+    }
+
+    /**
+     * We define the requirements to be an ambiguous variant as: being an indel (one allele is empty) and the non empty
+     * allele ends with the same nucleotide as what is in the reference assembly right before the variant.
+     *
+     * The reason for this is because we know the variants being processed have removed the leftmost nucleotide as
+     * redundant context, and we want it to have removed the rightmost nucleotide. This only yield different results
+     * if the original context nucleotide appears in the beginning and end of the other allele.
+     *
+     * Examples: let's say in position 10 there's AC, and an insertion happens between bases 10 and 11 of GGT.
```

Comment:
Maybe drop the "C"? I see what you mean but because it is not mentioned any more it may be somewhat confusing.